### PR TITLE
Feature/53 merge client server

### PR DESCRIPTION
**Pull request by CedricCORGE**

### CODE REVIEW

File: linux_installer.sh [hunk 1]
```diff
@@ -34,4 +34,4 @@ then
 fi
 
 echo -e "\033[0;32mBUILDING PROJECT\033[0m"
-mkdir -p build; cd build && conan install .. --build=missing  -s compiler.libcxx=libstdc++11 -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=True && cmake -DCMAKE_BUILD_TYPE=Debug .. && cmake --build  . -- -j 3
```
Comment (JohanCDev) [inline linux_installer.sh:37]: Leave it to 11

**Review by JohanCDev — CHANGES_REQUESTED** [review]

File: src/ECS/Components/GameTeamComponent.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+struct GameTeamComponent {
+
+    GameTeam team;
+
+    GameTeamComponent(GameTeam entity_team) : team(entity_team) {}
+
+    GameTeamComponent() : team(GameTeam::NONE) {}
+
+};
```
Comment (tbellicha) [inline src/ECS/Components/GameTeamComponent.hpp:24]: Missing new line at the EOF

File: src/ECS/Systems/AllSystem.hpp [hunk 3]
```diff
@@ -12,9 +12,10 @@
...
 
 #include "../World.hpp"
+#include "../../server/server.hpp"
 
 int drawable_system(World &world);
 int controllable_system(World &world, sf::Event event);
-int shooting_system(World &world);
+int shooting_system(World &world, NetworkServer &server);
 int velocity_system(World &world);
 int wave_system(World &world);
```
Comment (tbellicha) [inline src/ECS/Systems/AllSystem.hpp:21]: Missing new line at EOF

File: src/ECS/Systems/WaveSystem.cpp [hunk 4]
```diff
@@ -15,12 +15,12 @@ static float elapsed_time = 0.0;
...
 
     if (world.getClock().getElapsedTime().asSeconds() > (elapsed_time + 3.0)) {
         elapsed_time += 3.0;
-        random_y = std::rand() % world.getWindow().getSize().y;
-        world.create_enemy("assets/r-typesheet39.gif", Vector4i{34, 2, 64, 64}, Vector2f(1.0, 1.0), Vector2i(900, random_y), Vector2i(-2, 0), 0.04, world.getClock().getElapsedTime().asSeconds());
+        random_y = static_cast<float>(std::rand() % world.getWindow().getSize().y);
+        world.create_enemy(GameObject::ENEMY, Vector2f{900.0f, random_y}, Vector2i{-2, 0}, 0.04f, world.getClock().getElapsedTime().asSeconds());
     }
     return (0);
 }
```
Comment (tbellicha) [inline src/ECS/Systems/WaveSystem.cpp:26]: Missing new line at EOF

File: src/ECS/World.hpp [hunk 5]
```diff
@@ -1,117 +1,110 @@
...
 
 
     private:
         registry _r;
         ResourcesManager _manager;
         sf::RenderWindow _window;
         sf::Clock _clock;
+        std::map<GameObject, DrawableComponent> _drawMap;
+        Vector2i _player_direction;
 };
```
Comment (tbellicha) [inline src/ECS/World.hpp:110]: Missing new line at EOF

File: src/client/client.cpp [hunk 6]
```diff
@@ -109,3 +107,134 @@ void NetworkClient::run_service()
...
+    {GameMessage::S2C_GAME_END, game_end},
+    {GameMessage::S2C_MOVEMENT, movement},
+    {GameMessage::S2C_PLAYER_HIT, player_hit},
+    {GameMessage::S2C_OK, ok_packet},
+};
+
+void NetworkClient::processMessage(Message<GameMessage> &msg, World &world)
+{
+    (mapFunc[msg.header.id])(world, msg);
+}
```
Comment (tbellicha) [inline src/client/client.cpp:240]: Missing new line

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by Djadanis — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Review by JohanCDev — APPROVED** [review]

**Pull request merged**